tests/integration/olp-cpp-sdk-dataservice-read/VolatileLayerClientCacheTest.cpp

```diff
@@ -135,13 +122,19 @@ TEST_P(VolatileLayerClientCacheTest, GetVolatilePartitionsExpiry) {
   ASSERT_EQ(1u, partitions_response.GetResult().GetPartitions().size());
 
   // wait for the layer to expire in cache
-  std::this_thread::sleep_for(std::chrono::seconds(2));
-  request.WithFetchOption(olp::dataservice::read::OnlineIfNotFound);
-  future = catalog_client->GetPartitions(request);
-  partitions_response = future.GetFuture().get();
-
-  ASSERT_TRUE(partitions_response.IsSuccessful())
-      << ApiErrorToString(partitions_response.GetError());
+  bool expired = false;
+  for (;;) {
```

Comment:
```suggestion
  while (true) {
```
Prefer a `while-statement` to a `for-statement` when there is no obvious loop variable